Core/Model/ArticuloCombinacion.php

```diff
@@ -175,39 +171,198 @@ public function clear()
 
     public function test()
     {
-        if ($this->codigo === null) {
-            $this->codigo = (string) $this->getNewCodigo();
+        $sql = 'SELECT * FROM ' . static::tableName() . ' WHERE codigo = ' . $this->dataBase->var2str($cod) . ';';
+        $data = $this->dataBase->select($sql);
+        if (!empty($data)) {
+            return new self($data[0]);
         }
+
+        return false;
     }
 
     /**
-     * Devuelve un nuevo código para una combinación de artículo
+     * Remove all combinations of the item with reference = $ ref
      *
-     * @return int
+     * @param string $ref
+     *
+     * @return bool
      */
-    private function getNewCodigo()
+    public function deleteFromRef($ref)
     {
         $sql = 'SELECT MAX(' . $this->dataBase->sql2Int('codigo') . ') as cod FROM ' . $this->tableName() . ';';
         $cod = $this->dataBase->select($sql);
         if (!empty($cod)) {
             return 1 + (int) $cod[0]['cod'];
         }
+    }
 
-        return 1;
+    /**
+     * Returns an array with all the combinations of the item with reference = $ ref
+     *
+     * @param string $ref
+     *
+     * @return self[]
+     */
+    public function allFromRef($ref)
+    {
+        $lista = [];
+
+        $sql = 'SELECT * FROM ' . static::tableName() . ' WHERE referencia = ' . $this->dataBase->var2str($ref)
+            . ' ORDER BY codigo ASC, nombreatributo ASC;';
+        $data = $this->dataBase->select($sql);
+        if (!empty($data)) {
+            foreach ($data as $d) {
+                $lista[] = new self($d);
+            }
+        }
+
+        return $lista;
     }
 
     /**
-     * Devuelve un array con todos los datos de la combinación con código = $cod,
-     * ten en cuenta que lo que se almacenan son los pares atributo => valor.
+     * Returns an array with all the data of the combination with code = $ cod,
+     * keep in mind that what are stored are the attribute pairs => value.
      *
      * @param string $cod
      *
      * @return self[]
      */
     public function allFromCodigo($cod)
     {
-        $where = [new DataBaseWhere('codigo', $cod)];
```

Comment:
Esta función debe ser:
$where = [new DataBaseWhere('codigo', $cod)];
        $order = ['nombreatributo' => 'ASC'];
        return $this->all($where, $order);